ONSdigital/ssdc-rm-docker-dev
# 257 Update healthcheck to work in JRE

# Motivation and Context

With the java images moving to the JRE instead of JDK base image, the current healthcheck no longer works. Instead we must now compile and package the healthcheck into a JAR so it can run in the JRE.

# What has changed



- Compile and package java healthcheck
- Add make targets to re-package after any changes
- Update rm services healthchecks to use jar
- Update README

# How to test?



Run `BRANCH_NAME=257-use-jre-base-image SKIP_TESTS=true ./checkout_and_build_pr.sh` to build and run all the JRE base image branches. On this branch, the docker compose healthchecks should still work.

# Links



https://trello.com/c/bN3vy0xn/257-change-java-images-to-use-jre-base-image-3
